src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -870,7 +876,22 @@ HRESULT CProfilerManager::SetEventMask2(_In_ DWORD dwEventMaskLow, _In_ DWORD dw
 {
     HRESULT hr = S_OK;
 
-    m_dwEventMaskHigh |= dwEventMaskHigh;
+    if (GetIsInInitialize())
```

Comment:
I think we also need to handle when GetIsInitializingInstrumentationMethod() is true here, but I would make a separate change for that. The issue of the high event mask being set directly by the instrumentation methods already exists (if it's an issue at all).

Reply:
We will also have to revisit this behavior for attach

---
In reply to: [380962438](https://github.com/microsoft/CLRInstrumentationEngine/pull/202#discussion_r380962438) [](ancestors = 380962438)